src/cpp/src/cache_manager.hpp

```diff
@@ -129,19 +135,26 @@ class CacheManager {
                 if (m_key_cache.size() > decoder_layer_id) {
                     ov::Coordinate end_key = m_key_cache[decoder_layer_id].get_shape();
                     ov::Coordinate end_value = m_value_cache[decoder_layer_id].get_shape();
-
-                    key_roi_size_byte = m_key_cache[decoder_layer_id].get_byte_size();
-                    value_roi_size_byte = m_value_cache[decoder_layer_id].get_byte_size();
-                    key_cache_roi_end = static_cast<unsigned char*>(key_cache.data()) + key_roi_size_byte;
-                    value_cache_roi_end = static_cast<unsigned char*>(value_cache.data()) + value_roi_size_byte;
-                    
                     // copy current cache data
-                    ov::Tensor dst_key_roi(key_cache, start_key, end_key);
-                    ov::Tensor dst_value_roi(value_cache, start_value, end_value);
-
-                    m_key_cache[decoder_layer_id].copy_to(dst_key_roi);
-                    m_value_cache[decoder_layer_id].copy_to(dst_value_roi);
-
+                    if (key_precision == ov::element::u4) {
+                        size_t key_roi_size_byte = std::accumulate(end_key.begin(), end_key.end(), 1, std::multiplies<size_t>()) / sub_byte_data_type_multiplier(key_precision);
```

Comment:
what if `std::accumulate` returns odd value?
I suppose we need to round to upper value

Reply:
round up to nearest even above.